src/hooks/useValidateRepository.ts

```diff
@@ -0,0 +1,188 @@
+import {useCallback} from 'react';
+import {useDebounce} from 'react-use';
+
+import {NamePath} from 'antd/lib/form/interface';
+
+import {dummySecret} from '@utils/sources';
+
+const fieldsNames: string[] = ['uri', 'token', 'username', 'branch', 'path'];
+
+const authSearchString = 'username or token:';
+const uriSearchString = 'uri:';
+const branchSearchString = 'branch:';
+const pathSearchString = 'path:';
+const endSearchString = ', context';
+
+const getErrorMessage = (rawErrorString: string, searchString: string) =>
+  rawErrorString.slice(
+    rawErrorString.search(searchString) + searchString.length,
+    rawErrorString.search(endSearchString)
+  );
+
+const useValidateRepository = (
+  getFieldValue: (name: NamePath) => string,
```

Comment:
Why `getFieldValue` instead of the values object?

Reply:
No particular reason, do you think using values object is better?

Reply:
On the one hand, it may be simpler and more type-safe, on the other - it may be harder to compare them. Not sure what should we prefer, so will mark as resolved